File: java64/nivel5/exer51.java
import javax.swing.JOptionPane;

public class exer51 {
    public static void main(String[] args) {
        int n, soma = 0;
        int media = 0;
        int maior = 0;
        boolean menor = true;
        n = Integer.parseInt(JOptionPane.showInputDialog("Quantidade de alunos:"));
        int[] alunos = new int[n];

        for (int i = 0; i < n; i++) {
            alunos[i] = Integer.parseInt(JOptionPane.showInputDialog("Nota do "+ (i + 1) + "° aluno:"));
        }

        for (int g = 0; g < n; g++) {
            soma += alunos[g];
            media = soma / n;

            if (alunos[g] > 7) maior++;
            if (alunos[g] > 5)
                menor = false;
        }

        if (menor = false) {
            JOptionPane.showMessageDialog(null, "Meida: " + media + "\nMaior que 7: " + maior);
        } else {
            JOptionPane.showMessageDialog(null, "Não há nenhum aluno com nota acima de 5.");
        }
        
    }
}

File: pyhton64/nivel5/exer51.py
alunos_numero = int(input('Informe a quantidade de alunos: '))
notas = []
maior = 1
menor = 1

for i in range(1, alunos_numero + 1):
    notas.append(float(input('Notas: ')))

for x in range(len(notas)):
    if notas[x] > 7:
        maior += maior
        strCima = f'Quantidade de notas maiores que 7: {maior}'
    elif notas[x] <= 5:
        menor += menor
        strBaixo = f'Não há nenhum aluno com nota acima de 5'

media = sum(notas) / alunos_numero

for i in range(len(notas)):
    if notas[i] > 7:
        print(f'{strCima}\nMédia: {media}')
    elif notas[i] <= 5:
        print(f'{strBaixo}\nMédia: {media}')